catalog/admin/admin-poller-service/src/main/java/org/codice/ddf/catalog/admin/poller/AdminPollerServiceBean.java

```diff
@@ -215,6 +249,86 @@ public int compare(Map<String, Object> o1, Map<String, Object> o2) {
+                .is()
+                .equalTo()
+                .text(source);
+        Query query = new QueryImpl(filter);
+
+        QueryResponse queryResponse = catalogFramework.query(new QueryRequestImpl(query));
+        List<Result> metacards = queryResponse.getResults();
+        if (metacards != null && metacards.size() > 0) {
+            Metacard metacard = metacards.get(0)
+                    .getMetacard();
+            if (metacard != null) {
+                List<Serializable> currentlyPublishedLocations = metacard.getAttribute(
+                        RegistryObjectMetacardType.PUBLISHED_LOCATIONS)
+                        .getValues();
+
+                List<String> publishLocations = destinations;
+                List<String> unpublishLocations = new ArrayList<>();
+                List<Serializable> updatedPublishedLocations = new ArrayList<>();
+
+                unpublishLocations.addAll(currentlyPublishedLocations.stream()
+                        .filter(location -> !destinations.contains((String) location))
+                        .map(location -> (String) location)
+                        .collect(Collectors.toList()));
+
+                updatedPublishedLocations.addAll(destinations);
+
+                for (String id : publishLocations) {
+                    Filter existingMetacardsFilter = filterBuilder.attribute(
+                            RegistryObjectMetacardType.REGISTRY_ID)
+                            .is()
+                            .equalTo()
+                            .text(metacard.getAttribute(RegistryObjectMetacardType.REGISTRY_ID)
+                                    .getValue()
+                                    .toString());
+
+                    Query existingMetacardsQuery = new QueryImpl(existingMetacardsFilter);
+                    SourceResponse existingQueryResponse = catalogStoreMap.get(id)
+                            .query(new QueryRequestImpl(existingMetacardsQuery));
+                    List<Result> existingMetacards = existingQueryResponse.getResults();
+
+                    if (existingMetacards.size() == 0) {
+                        CreateRequest createRequest = new CreateRequestImpl(metacard);
+                        try {
+                            catalogStoreMap.get(id)
+                                    .create(createRequest);
+                        } catch (IngestException e) {
+                            updatedPublishedLocations.remove(id);
+                            LOGGER.error("Unable to ingest metacard", e.getMessage());
+                        }
+                    }
+                }
+                for (String id : unpublishLocations) {
+                    DeleteRequest deleteRequest = new DeleteRequestImpl(metacard.getId());
+                    try {
+                        catalogStoreMap.get(id)
+                                .delete(deleteRequest);
+                    } catch (IngestException e) {
```

Comment:
What if the metacard was already deleted? Possible extra query to check if the metacard is present after attempt.